src/tutorials/gno.land/gbe/07-loops/index.ts

```diff
@@ -2,36 +2,7 @@ const markdownContent: string = `
 
 Loops works as in standard Go, but you have to be careful to always check for the right condition to break it, if not your application will consume all the available Gas.
 
-\`\`\`go
-const totalLoops = 10
-
-func forLoop() string {
-	var out string
-	for i := 0; i < totalLoops; i++ {
-		out = out + fmt.Sprintf("%d", i)
-	}
-
-	return out
-}
-
-func whileLoop() string {
-	var loops int
-	var out string
-	for {
-		if loops == totalLoops {
-			break
-		}
-
-		out = out + "A"
-
-		// you have to be sure to reach the break condition,
-		// if not you will exaust all the available gas for this execution
-		loops++
-	}
-
-	return out
-}
-\`\`\`
+<Playground open="src/tutorials/gno.land/gbe/07-loops/loops.gno" files="%5B%7B%22path%22%3A%22src%2Ftutorials%2Fgno.land%2Fgbe%2F07-loops%2Floops.gno%22%2C%22content%22%3A%22const%20totalLoops%20%3D%2010%5Cn%5Cnfunc%20forLoop()%20string%20%7B%5Cn%5Ctvar%20out%20string%5Cn%5Ctfor%20i%20%3A%3D%200%3B%20i%20%3C%20totalLoops%3B%20i%2B%2B%20%7B%5Cn%5Ct%5Ctout%20%3D%20out%20%2B%20fmt.Sprintf(%5C%22%25d%5C%22%2C%20i)%5Cn%5Ct%7D%5Cn%5Cn%5Ctreturn%20out%5Cn%7D%5Cn%5Cnfunc%20whileLoop()%20string%20%7B%5Cn%5Ctvar%20loops%20int%5Cn%5Ctvar%20out%20string%5Cn%5Ctfor%20%7B%5Cn%5Ct%5Ctif%20loops%20%3D%3D%20totalLoops%20%7B%5Cn%5Ct%5Ct%5Ctbreak%5Cn%5Ct%5Ct%7D%5Cn%5Cn%5Ct%5Ctout%20%3D%20out%20%2B%20%5C%22A%5C%22%5Cn%5Cn%5Ct%5Ct%2F%2F%20you%20have%20to%20be%20sure%20to%20reach%20the%20break%20condition%2C%5Cn%5Ct%5Ct%2F%2F%20if%20not%20you%20will%20exaust%20all%20the%20available%20gas%20for%20this%20execution%5Cn%5Ct%5Ctloops%2B%2B%5Cn%5Ct%7D%5Cn%5Cn%5Ctreturn%20out%5Cn%7D%22%7D%5D">
```

Comment:
we should keep a way to declare source codes more like it was the case previously.

what about creating a real `loops.gno` file in git, and then find a way to inject it?  this way we'll have real .gno files that we can run with the `gno binary` too.

Reply:
I think what you are describing is already case, please [see](https://github.com/gnolang/gno-by-example/tree/feat/playground/src/tutorials/gno.land/gbe/07-loops), gno files are there.

`index.ts` contains the generated code from the gno files by using the meta information in the readme file. I believe the generated file doesn't need to be human readable, right?

Reply:
@ilgooz is right

This `index.ts` is something generated by a script, basically glue code we import when the user wants to see a specific tutorial, so it doesn't necessarily need to be human readable.

The tutorials (gno files) themselves are untouched, and in fact we run full on tests for them in the CI to make sure they are valid and passing

Just for comparison, Solidity by Example encodes these sorts of glue code imports in base64

Reply:
Ok, sounds good. Thank you for the clarification.

Please, ping me for the next PR adding an exercise, so I can check if the review looks reasonable.

Isn't it a good practice to name generated files with `.gen.ts`? We could set this extension as `linguist-generated` in `.gitattributes` to further ease the review process.

Reply:
Ok, sounds good. Thank you for the clarification.

Please, ping me for the next PR adding an exercise, so I can check if the review looks reasonable.

Isn't it a good practice to name generated files with `.gen.ts`? We could set this extension as `linguist-generated` in `.gitattributes` to further ease the review process.